### ISSUE TITLE
User story - Job Post apply

### ISSUE THREAD

**Issue by stephendawsondev** [stephendawsondev/StackPortfolio#37]
### If applicable, add a related Epic.

Epic: #28 

### User Story

As a **tech user**, I can **apply to a job posting** so that I can **get a job**.

### Acceptance Criteria

- There should be a button that the user can click to apply
- The user can add some details such as a cover letter
- The recruiter is notified of the application and user's profile is shared

### Please ensure you've completed these tasks before marking the issue as done.

- [ ] Reviewed the user story for clarity and completeness.
- [ ] Identified and linked related issues or epics.
- [ ] Created or updated the necessary documentation.
- [ ] Written unit and/or integration tests for functionality.
- [ ] All code changes have been manually tested to ensure they meet the acceptance criteria.
- [ ] Updated the issue with any relevant notes or feedback (to date).
- [ ] Merged into the main branch and deployed to production (if applicable).

### Please add any further tasks that need to be completed before the issue can be marked as done.

- [ ] Add an application view
